### DISCUSSION TITLE
Codehilte custom pygment lexer not working

### DISCUSSION THREAD

**Discussion by raikoug** [mkdocs/mkdocs#2945] [Q&A] [answered]
Hi,

After hours of searching, writing, testing, I'm at a dead end.  

My lexer works:
```bash
python -m pygments -O full -f html -l ios -o test.html test_input.txt
```
creates a perfect test.html file with the correct highlight.  
I did install my pygment with setutools: 

```bash
python -m setup.py develop
```
The lexer is listed correctly:
```bash
pygmentize -L lexers
...
* ios, IOS configuration file:
    Cisco IOS (filenames *.ios)
...
```

I'm using 
```yaml
markdown_extensions:
  - codehilite
```
in my mkdocs.yml, but there is not my highlight.  
Aliases are correct.

CodeHilite alone it's working like a charm:

```python
# code from codeHilite homepage to test it
from pygments.formatters import HtmlFormatter
from markdown.extensions.codehilite import CodeHiliteExtension


class CustomHtmlFormatter(HtmlFormatter):
    def __init__(self, lang_str='', **options):
        super().__init__(**options)
        # lang_str has the value {lang_prefix}{lang}
        # specified by the CodeHilite's options
        self.lang_str = lang_str

    def _wrap_code(self, source):
        yield 0, f'<code class="{self.lang_str}">'
        yield from source
        yield 0, '</code>'


open('test.html', 'w').write(markdown.markdown(open('test_input.md').read(), extensions=[CodeHiliteExtension(pygments_formatter=CustomHtmlFormatter)],))
```
Generates a perfect html with my highlists.

Help

**Comment by facelessuser** [comment, marked as answer]
You shouldn't need a custom formatter if providing a custom lexer. You simply need your lexer to properly define the proper entry point and have that lexer installed in your environment. If done correctly, you should be able to use the language code in  your fenced code blocks.

I don't use CodeHilite, but instead, use `pymdownx.superfences`, but I imagine the behavior should be the same in regards to custom lexers. I personally use custom lexers myself in conjunction with MkDocs and experience no issues.

**Reply by raikoug** [reply]
Hi, As soon I used 

```yaml
markdown_extensions:
  - pymdownx.highlight:
      anchor_linenums: true
  - pymdownx.superfences 
```

It worked. I now think it is a CodeHilite extension bug...

**Reply by facelessuser** [reply]
That would be surprising, but I guess not outside the realm of possibilities.
